src/Hive/Startup.cs

```diff
@@ -119,12 +145,30 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             _ = app.UsePathBase(Configuration.GetValue<string>("PathBase"))
                 .UseSerilogRequestLogging()
-                .UseHttpsRedirection()
-                .UseRouting()
-                .UseAuthentication()
+                .UseRouting();
+
+            var web = Configuration.GetSection(WebOptions.ConfigHeader);
+            if (web.Exists())
+            {
+                var webOptions = web.Get<WebOptions>();
+                if (webOptions.HTTPSRedirection ?? true)
+                    _ = app.UseHttpsRedirection();
+                if (webOptions.CORS)
+                {
+                    _ = app.UseCors(webOptions.PolicyName);
```

Comment:
What exactly does CORS configuration do in ASP.NET? I assume it just sets some headers, but which ones?

Reply:
CORS works by the client sending a request to the server with the `OPTIONS` method, then if that passes, the client sends the real request with the right method. Headers like `Access-Control-Allow-Headers`, `Access-Control-Allow-Methods`, `Access-Control-Allow-Origin`, and related get set for all requests with the `OPTIONS` method. That's primarily what this does.